app/liquid_tags/link_tag.rb

```diff
@@ -27,16 +27,28 @@ def render(_context)
 
   private
 
-  def parse_url_for_article(url)
-    url = ActionController::Base.helpers.strip_tags(url)
-    hash = Rails.application.routes.recognize_path(url)
+  def parse_url_for_article(slug_or_path_or_url)
+    slug_or_path_or_url = ActionController::Base.helpers.strip_tags(slug_or_path_or_url).strip
+
+    hash = article_hash(slug_or_path_or_url)
+
+    raise_error if hash.nil?
+
     article = find_article_by_user(hash) || find_article_by_org(hash)
     raise_error unless article
     article
   rescue StandardError
     raise_error
   end
 
+  def article_hash(slug_or_path_or_url)
+    path = Addressable::URI.parse(slug_or_path_or_url).path
+    path.slice!(0) if path.starts_with?("/") # remove leading slash if present
+    path.slice!(-1) if path.ends_with?("/") # remove trailing slash if present
```

Comment:
alternatively these 2 lines could be shortened to: 

```rb
path.gsub!(/\A\/|\/\z/, '') # remove leading and trailing slashes if present
```

Reply:
@NARKOZ  the reason why I chose slice is duplicitous:

* it's clearer (especially if the comment disappears)
* it's slightly faster: https://stackoverflow.com/a/3614592/4186181

I thought it was a good compromise between speed and readability